brushfire-training/src/main/scala/com/stripe/brushfire/local/CrossEntropyExample.scala

```diff
@@ -0,0 +1,44 @@
+package com.stripe.brushfire.local
+
+import com.stripe.bonsai._
+import com.stripe.brushfire._
+import com.twitter.algebird._
+
+import AnnotatedTree.{AnnotatedTreeTraversal, fullBinaryTreeOpsForAnnotatedTree}
+
+object CrossEntropyExample extends Defaults {
+
+  def main(args: Array[String]) {
+    val path = args.head
+    val cols = args.tail.toList
+
+    // T = Map[String, Long]
+    // P = String
+    // E = Double
+
+    val trainingData = Lines(path).map { line =>
+      val parts = line.split(",").reverse.toList
+      val label = parts.head
+      val values = parts.tail.map { s => s.toDouble }
+      Instance(line, 0L, Map(cols.zip(values): _*), Map(label -> 1L))
+    }.toIterable
+
+    val totalData: Map[String, Long] = Monoid.sum(trainingData.iterator.map(_.target))
```

Comment:
Agreed that we should do that (that's what I meant in https://github.com/stripe/brushfire/pull/93/files/6aa9b7fe7adaaf9dbbce7f79abd767e56c4f2ecd#r68490840). I take your point about `Error` needing a `present` function. `Error[T,P,E]` could equally be `Aggregator[(T,P),E,E2]` where `E2` is the thing you actually care about. Or you could decide you just cared about having an `Ordering[E]` (which comes up a lot in practice) and which would do the transformation to `Double` or whatever internally.